src/components/material/ContextualHints.js

```diff
@@ -0,0 +1,94 @@
+const MissionUtils = require("@woowacourse/mission-utils");
+
+class ContextualHints {
+  constructor(computerNum, playerNum, GivePlayerHint) {
+    this.computerNum = computerNum;
+    this.playerNum = playerNum;
+    this.NumOfSamePosition = playerNum
+      ?.split("")
+      .filter((num, index) => +num === +computerNum[index]).length;
+    this.NO_STRIKE = 0;
+    this.ONE_STRIKE = 1;
+    this.TWO_STRIKE = 2;
+    this.THREE_STRIKE = 3;
+    this.NO_BALL = 0;
+    this.ONE_BALL = 1;
+    this.TWO_BAll = 2;
+    this.THREE_BALL = 3;
+    this.RESTART = "1";
+    this.GAVE_OVER = "2";
+    this.GivePlayerHint = GivePlayerHint;
+  }
+
+  HowManyEqualNum() {
+    const isSame = this.playerNum
+      .split("")
+      .map((num) => this.computerNum.includes(+num));
+    return isSame.filter((value) => Boolean(value)).length;
+  }
+
+  endGame() {
+    MissionUtils.Console.readLine(
+      "3개의 숫자를 모두 맞히셨습니다. 게임 종료 \n" +
+        "게임을 새로 시작하려면 1, 종료하려면 2를 입력하세요.",
+      (answer) => {
+        console.log(answer);
+        if (answer === this.RESTART) {
+          const givePlayerHint = new this.GivePlayerHint();
+          givePlayerHint.runGame();
+        }
```

Comment:
숫자를 다 못맞춰서 다시하는 경우에는 밑의 코드들을 굳이 실행시켜주지 않아도 될 것 같아요. 그래서 바로 `runGame()`을 return 해줘도 좋았을 것 같아요!

```suggestion
if (answer === this.RESTART) {
          const givePlayerHint = new this.GivePlayerHint();
         return givePlayerHint.runGame();
```